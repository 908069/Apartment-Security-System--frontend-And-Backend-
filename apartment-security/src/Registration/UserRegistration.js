import React ,{useState,useEffect}from 'react'
import axios from 'axios'


    function UserRegistration(props) {
    const [userregistration, setuserregistration] = useState({
        username :"",
        password :"",
        uniqueKey :"",
        mobilenumber :"",
        email:"",
        role :"",
        msg:""
        
    })
    
    const [role, setrole] = useState()
    
    const handleInput=(e)=>{
       const name =e.target.name
       const value=e.target.value
       setuserregistration({...userregistration,[name] : value})
        
    }
  
   const handleSubmit=(e)=>{
        e.preventDefault()
        const newRecord={...userregistration}
        let data = {
                "emailId": newRecord.email,
                "mobileNo": newRecord.mobilenumber,
                "name": newRecord.username,
                "password": newRecord.password,
                "role": newRecord.role,
                "userId": newRecord.uniqueKey
        }
        



        axios.get('http://localhost:8082/springfox/api/user/users')
        .then(response=>{
           if(response.status == 200){
            const check2= response.data.filter(user=>user.userId===data.userId);
           const check1= response.data.filter(user=>user.emailId===data.emailId);
           
           console.log(check2)
         // console.log(check1);
          if(check1.length!=0||check2.length!=0){
              setuserregistration({...userregistration,msg : 'Email is already exist'});
          }
          else if(check2.length!=0){
            setuserregistration({...userregistration,msg : ' Unique Key is already exist'});
          }
          else{
              console.log(data)
            axios.post('http://localhost:8082/springfox/api/user/users',data)
            .then(response=>{
               console.log(response)
               
               if(response.status == 200){
              
                  console.log(userregistration.role)
                  if(userregistration.role=='owner'){
                      props.history.push('/addflatdetails')
                   }
                   else if(userregistration.role=='guard'){
                     props.history.push('/addGuardShift')
                   }}
              else{
                   alert("Invalid UserName or Password")
                  
               }
            }).then(response=>{
                if(response){
                    props.history.replace('/addflatdetails')
                }
            }
              )
            .catch(error=>{
               console.log(error)
           })
          }
          
          }
           else{
              this.setState({...userregistration,msg : 'Authentication error'});
              document.getElementById("showSuccessResponse").innerHTML = "An error occurred while uploading the data."
           }
        })
        .catch(error=>{
           

           console.log(error)
       })

     

    }
    return (
        <div>
            <h1>Registration</h1>
            <form action ="" onSubmit={handleSubmit}>
                <label htmlFor="username" 
                >Name</label>
                <br></br>
                <input type="text" autoComplete="off" onChange={handleInput}
                 value={userregistration.username} name="username" id="username" required/>
                <br></br>
                <label htmlFor="password" 
                > Password</label>
                <br></br>
                <input type="password" autoComplete="off" onChange={handleInput}
                 value ={userregistration.password} name="password" id="password" required/>
                <br></br>
                <label htmlFor="mobilenumber" 
                > Mobile Number</label>
                <br></br>
                <input type="tel" autoComplete="off" onChange={handleInput}
                 value ={userregistration.mobilenumber} name="mobilenumber" id="mobilenumber" 
                pattern="[0-9]{10}" required/>
                <br></br>
                <label htmlFor="uniqueKey" 
                > Unique Key</label>
                <br></br>
                <input type="tel" autoComplete="off" onChange={handleInput} 
                value ={userregistration.uniqueKey} name="uniqueKey" id="uniqueKey" 
                pattern="[0-9]{5}" required/>
                <br></br><br></br>
                <label htmlFor="email" 
                >email</label>
                <br></br>
                <input type="email" id="email"  onChange={handleInput} 
                value ={userregistration.email} autoComplete="off" name="email" required></input>
                <br></br><br></br>
                <label> Owner:</label>
                <input type="radio"  
                onChange={() =>setrole("owner"),handleInput} name="role" value="owner" required/>
                <label> Guard:</label>
                <input type="radio" 
                onChange={()=> setrole("guard"),handleInput} name="role" value="guard" />
                <br></br>
                <br></br>
                
                <button type="submit" > Next</button>
            </form>
            <h1>{userregistration.msg}</h1>
<div id="showSuccessResponse"></div>
        </div>
    )}
export default UserRegistration
