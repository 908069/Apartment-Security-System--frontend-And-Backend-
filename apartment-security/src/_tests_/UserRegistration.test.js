import React from 'react';
import UserRegistration from '../Registration/UserRegistration';
import renderer from 'react-test-renderer';
 
it('renders correctly', () => {
 const tree = renderer
   .create(<UserRegistration/>)
   .toJSON();
   console.log(tree)
   expect(tree).toMatchSnapshot();
});