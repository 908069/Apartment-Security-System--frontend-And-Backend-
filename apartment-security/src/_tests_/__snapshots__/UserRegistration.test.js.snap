// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="card col-md-6 offset-md-3 offset-md-3"
    >
      <h3>
        Registration
      </h3>
      <form
        action=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="username"
          >
            Name
          </label>
          <br />
          <input
            autoComplete="off"
            id="username"
            name="username"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <br />
          <label
            htmlFor="password"
          >
             Password
          </label>
          <br />
          <input
            autoComplete="off"
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
          <br />
          <label
            htmlFor="mobilenumber"
          >
             Mobile Number
          </label>
          <br />
          <input
            autoComplete="off"
            id="mobilenumber"
            name="mobilenumber"
            onChange={[Function]}
            pattern="[0-9]{10}"
            required={true}
            type="tel"
            value=""
          />
          <br />
          <label
            htmlFor="uniqueKey"
          >
             Unique Key
          </label>
          <br />
          <input
            autoComplete="off"
            id="uniqueKey"
            name="uniqueKey"
            onChange={[Function]}
            pattern="[0-9]{5}"
            required={true}
            type="tel"
            value=""
          />
          <br />
          <label
            htmlFor="email"
          >
            email
          </label>
          <br />
          <input
            autoComplete="off"
            id="email"
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
          <br />
          <br />
          <label>
             Owner:
          </label>
          <input
            name="role"
            onChange={[Function]}
            required={true}
            type="radio"
            value="owner"
          />
          <label>
             Guard:
          </label>
          <input
            name="role"
            onChange={[Function]}
            type="radio"
            value="guard"
          />
          <br />
          <br />
          <button
            class="btn btn-outline-success my-2 my-sm-0"
            type="submit"
          >
             Next
          </button>
        </div>
      </form>
      <h6>
        
      </h6>
    </div>
  </div>
  <div
    id="showSuccessResponse"
  />
</div>
`;
